### PR TITLE
Static link and cleanup the manifest

### PR DESCRIPTION
**Pull request by hfiguiere**
This would have come up in the review. I will comment things inline.

### CODE REVIEW

File: org.freedesktop.LinuxAudio.Plugins.ZynFusion.json [hunk 1]
```diff
@@ -6,43 +6,48 @@
...
+        "prepend-path": "/app/extensions/Plugins/ZynFusion/bin",
+        "prepend-ld-library-path": "/app/extensions/Plugins/ZynFusion/lib",
         "prepend-pkg-config-path": "/app/extensions/Plugins/ZynFusion/lib/pkgconfig"
     },
     "modules": [
-        "shared-modules/linux-audio/liblo.json",
+        "shared-modules/linux-audio/liblo-static.json",
         "shared-modules/linux-audio/lv2.json",
-        "python-2.7-build-only.json",
+        "shared-modules/python2.7/python-2.7.json",
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.ZynFusion.json:24]: It is recommended to use the shared-module for python 2.7. See above (lines 9-14) for the cleanup.

File: org.freedesktop.LinuxAudio.Plugins.ZynFusion.json [hunk 2]
```diff
@@ -6,43 +6,48 @@
...
+    ],
     "build-options": {
         "prefix": "/app/extensions/Plugins/ZynFusion",
+        "prepend-path": "/app/extensions/Plugins/ZynFusion/bin",
+        "prepend-ld-library-path": "/app/extensions/Plugins/ZynFusion/lib",
         "prepend-pkg-config-path": "/app/extensions/Plugins/ZynFusion/lib/pkgconfig"
     },
     "modules": [
-        "shared-modules/linux-audio/liblo.json",
+        "shared-modules/linux-audio/liblo-static.json",
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.ZynFusion.json:22]: For plugins we should use the static variant as to not conflict with any other version.

File: org.freedesktop.LinuxAudio.Plugins.ZynFusion.json [hunk 3]
```diff
@@ -6,43 +6,48 @@
...
-          "buildsystem": "simple",
-          "build-commands": [
-                "./configure --prefix=$FLATPAK_DEST",
-                "make",
-                "make install"
-          ],
-          "build-options": {
-            "env": {
-                    "PREFIX": "${FLATPAK_DEST}",
-                    "DESTDIR": "${FLATPAK_DEST}"
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.ZynFusion.json:28]: You don't need these.

File: org.freedesktop.LinuxAudio.Plugins.ZynFusion.json [hunk 4]
```diff
@@ -6,43 +6,48 @@
...
-        "python-2.7-build-only.json",
+        "shared-modules/python2.7/python-2.7.json",
         {
-          "name": "libmxml",
-          "buildsystem": "simple",
-          "build-commands": [
-                "./configure --prefix=$FLATPAK_DEST",
-                "make",
-                "make install"
-          ],
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.ZynFusion.json:24]: `"buildsystem": "autotools"`, and you remove the build commands and the `build-options` below

File: org.freedesktop.LinuxAudio.Plugins.ZynFusion.json [hunk 5]
```diff
@@ -6,43 +6,48 @@
...
-          ]
+            "name": "libmxml",
+            "buildsystem": "autotools",
+            "build-options": {
+                "env": {
+                    "CFLAGS": "-fPIC"
+                }
+            },
+            "config-opts": [
+                "--disable-shared"
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.ZynFusion.json:34]: CFLAGS needed to link static

And config-opts to remove shared libraries at build time.

This to link mxml statically, which is recommended.

File: org.freedesktop.LinuxAudio.Plugins.ZynFusion.json [hunk 6]
```diff
@@ -59,12 +64,8 @@
             ],
             "build-options": {
                 "env": {
-                    "PREFIX": "${FLATPAK_DEST}",
-                    "DESTDIR": "${FLATPAK_DEST}",
                     "VERSION": "3.0.5",
-                    "BUILD_MODE": "release",
-                    "PATH": "/app/bin:/usr/bin:/app/extensions/Plugins/ZynFusion/bin",
-                    "LD_LIBRARY_PATH": "/app/lib:/usr/lib:/app/extensions/Plugins/ZynFusion/lib"
+                    "BUILD_MODE": "release"
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.ZynFusion.json:68]: You don't need there. for `PATH` and `LD_LIBRARY_PATH` see the added lines 17-18.

File: org.freedesktop.LinuxAudio.Plugins.ZynFusion.json [hunk 7]
```diff
@@ -99,37 +100,33 @@
         },
         {
             "name": "zynadsubfx",
-            "buildsystem": "cmake",
+            "buildsystem": "cmake-ninja",
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.ZynFusion.json:103]: we prefer to use `cmake-ninja`

File: org.freedesktop.LinuxAudio.Plugins.ZynFusion.json [hunk 8]
```diff
@@ -99,37 +100,33 @@
...
             "config-opts": [
+                "-DPluginLibDir=${FLATPAK_DEST}",
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.ZynFusion.json:105]: This will tell where to put the plugin dirs. No symlinks needed.

File: org.freedesktop.LinuxAudio.Plugins.ZynFusion.json [hunk 9]
```diff
@@ -99,37 +100,33 @@
...
                 "-DGuiModule=zest",
                 "-DDemoMode=false"
             ],
+            "build-options": {
+                "env": {
+                    "LDFLAGS": "-lm"
+                }
+            },
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.ZynFusion.json:113]: we need this for static linking

File: org.freedesktop.LinuxAudio.Plugins.ZynFusion.json [hunk 10]
```diff
@@ -99,37 +100,33 @@
...
             "cleanup": [
-              "/lib/vst",
-              "/bin",
-              "/share"
+                "/bin",
+                "/share/bash-completion",
+                "/share/doc"
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.ZynFusion.json:117]: by removing `/share` you removed the sound banks and presets.

Also we'll keep the vst.

File: org.freedesktop.LinuxAudio.Plugins.ZynFusion.json [hunk 11]
```diff
@@ -99,37 +100,33 @@
...
-                "ln -sf ../lib/lv2/ZynAddSubFX.lv2 ${FLATPAK_DEST}/lv2/",
-                "ln -sf ../lib/lv2/ZynAddSubFX.lv2presets ${FLATPAK_DEST}/lv2/",
-                "ln -sf ../lib/lv2/ZynAlienWah.lv2 ${FLATPAK_DEST}/lv2/",
-                "ln -sf ../lib/lv2/ZynChorus.lv2 ${FLATPAK_DEST}/lv2/",
-                "ln -sf ../lib/lv2/ZynDistortion.lv2 ${FLATPAK_DEST}/lv2/",
-                "ln -sf ../lib/lv2/ZynDynamicFilter.lv2 ${FLATPAK_DEST}/lv2/",
-                "ln -sf ../lib/lv2/ZynEcho.lv2 ${FLATPAK_DEST}/lv2/",
-                "ln -sf ../lib/lv2/ZynPhaser.lv2 ${FLATPAK_DEST}/lv2/",
-                "ln -sf ../lib/lv2/ZynReverb.lv2 ${FLATPAK_DEST}/lv2/",
+                "mv ${FLATPAK_DEST}/vst ${FLATPAK_DEST}/lxvst",
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.ZynFusion.json:120]: sadly we need to rename the vst directory.

File: org.freedesktop.LinuxAudio.Plugins.ZynFusion.json [hunk 12]
```diff
@@ -99,37 +100,33 @@
...
             ],
             "sources": [
                 {
-                    "type": "git",
-                    "url": "https://github.com/zynaddsubfx/zynaddsubfx",
-                    "tag": "3.0.5",
-                    "commit": "4d4aedf834dbd13c6e5f07ac512c9da74732fd58"
+                    "type": "archive",
+                    "url": "https://download.sourceforge.net/sourceforge/zynaddsubfx/zynaddsubfx/3.0.5/zynaddsubfx-3.0.5.tar.bz2",
+                    "sha256": "7447322268114a1e0ac5f281ac37a09a78e761a7be61999caf79100049789f63"
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.ZynFusion.json:129]: I tend to prefer tarballs. It's faster (don't need to recheck git every time)

File: org.freedesktop.LinuxAudio.Plugins.ZynFusion.metainfo.xml [hunk 13]
```diff
@@ -3,7 +3,7 @@
   <id>org.freedesktop.LinuxAudio.Plugins.ZynFusion</id>
   <extends>org.freedesktop.LinuxAudio.BaseExtension</extends>
   <name>ZynAddSubFx Synthesizer</name>
-  <summary>ZynAddSubFx with Zyn-Fusion UI plugins</summary>
+  <summary>ZynAddSubFx (with Zyn-Fusion UI) LV2/VST plugins</summary>
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.ZynFusion.metainfo.xml:6]: just some minor adjustement, I like to have the plugin format directly in the summary. Works nice for search.

**Comment by hfiguiere** [comment]
Thank you so much for building the fusion UI. I had a version with ntk and the UI didn't start.

**Comment by takuvata** [comment]
Thank you so much! Building software is not something I normally do, so any help is more than welcome.

**Comment by hfiguiere** [comment]
I have that experience, which mean that I have already made all these mistakes. Happy to pass the knowledge on and help.

**Comment by takuvata** [comment]
Also have a problem with oscilloscope not showing anything, details in the issue: https://github.com/takuvata/org.freedesktop.LinuxAudio.Plugins.ZynFusion/issues/3

TL;DR: it works when zyn is built from master branch.
